api/dao/reaperutil.py

```diff
@@ -144,4 +157,51 @@ def create_container_hierarchy(metadata):
         upsert=True,
         return_document=pymongo.collection.ReturnDocument.AFTER,
     )
-    return ReapedAcquisition(acquisition_obj, file_)
+    return TargetAcquisition(acquisition_obj, file_)
+
+def update_container_hierarchy(metadata, acquisition_id, level):
+    project = metadata.get('project')
+    session = metadata.get('session')
+    acquisition = metadata.get('acquisition')
+    now = datetime.datetime.utcnow()
+    if acquisition.get('timestamp'):
+        acquisition['timestamp'] = dateutil.parser.parse(acquisition['timestamp'])
+    acquisition['modified'] = now
+    acquisition_obj = _update_container({'_id': acquisition_id}, acquisition, 'acquisitions')
+    if acquisition_obj is None:
+        raise APIStorageException('acquisition doesn''t exist')
+    if acquisition.get('timestamp'):
+        session_obj = config.db.sessions.find_one_and_update(
+            {'_id': acquisition_obj['session']},
+            {
+                '$min': dict(timestamp=acquisition['timestamp']),
+                '$set': dict(timezone=acquisition.get('timezone'))
+            },
+            return_document=pymongo.collection.ReturnDocument.AFTER
+        )
+        config.db.projects.find_one_and_update(
+            {'_id': session_obj['project']},
+            {
+                '$max': dict(timestamp=acquisition['timestamp']),
+                '$set': dict(timezone=acquisition.get('timezone'))
+            }
+        )
+    session_obj = None
+    if session:
+        session['modified'] = now
+        session_obj = _update_container({'_id': acquisition_obj['session']}, session, 'sessions')
+    if project:
+        project['modified'] = now
+        if not session_obj:
+            session_obj = config.db.sessions.find_one({'_id': acquisition_obj['session']})
+        _update_container({'_id': session_obj['project']}, project, 'projects')
```

Comment:
It looks like this function is called several times without error handling.
If mongo throws an exception halfway through, will the result be a partial update? How bad is that?